fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java

```diff
@@ -3702,82 +3706,101 @@ private List<List<String>> getTableList(List<MultipartIdentifierContext> ctx) {
+                            hints.add(new SelectHintLeading(hintName, leadingParameters));
+                            break;
+                        case "ordered":
+                            hints.add(new SelectHintOrdered(hintName));
+                            break;
+                        case "use_cbo_rule":
+                            List<String> useRuleParameters = new ArrayList<>();
+                            for (HintAssignmentContext kv : hintStatement.parameters) {
+                                if (kv.key != null) {
+                                    String parameterName = visitIdentifierOrText(kv.key);
+                                    useRuleParameters.add(parameterName);
+                                }
                             }
-                        }
-                        hints.add(new SelectHintUseCboRule(hintName, useRuleParameters, false));
-                        break;
-                    case "no_use_cbo_rule":
-                        List<String> noUseRuleParameters = new ArrayList<>();
-                        for (HintAssignmentContext kv : hintStatement.parameters) {
-                            String parameterName = visitIdentifierOrText(kv.key);
-                            if (kv.key != null) {
-                                noUseRuleParameters.add(parameterName);
+                            hints.add(new SelectHintUseCboRule(hintName, useRuleParameters, false));
+                            break;
+                        case "no_use_cbo_rule":
+                            List<String> noUseRuleParameters = new ArrayList<>();
+                            for (HintAssignmentContext kv : hintStatement.parameters) {
+                                String parameterName = visitIdentifierOrText(kv.key);
+                                if (kv.key != null) {
+                                    noUseRuleParameters.add(parameterName);
+                                }
                             }
+                            hints.add(new SelectHintUseCboRule(hintName, noUseRuleParameters, true));
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            }
+            newPlan = new LogicalSelectHint<>(hints.build(), newPlan);
+        }
+        if (!commonHintContexts.isEmpty()) {
+            for (ParserRuleContext hintContext : commonHintContexts) {
+                if (hintContext instanceof SelectHintContext) {
+                    SelectHintContext commonHintContext = (SelectHintContext) hintContext;
+                    if (commonHintContext.hintStatement != null && commonHintContext.hintStatement.hintName != null) {
+                        String text = commonHintContext.hintStatement.hintName.getText();
+                        if (text.equalsIgnoreCase("PREAGGOPEN")) {
+                            newPlan = new LogicalCommonHint<>(newPlan);
```

Comment:
If only support PREAGGOPEN, use LogicalPreAggOpenHint is better. Or let PreAggHint as member of LogicalCommonHint, it will be easier to support other hint type in future

Reply:
I plan to use LogicalCommonHint for other hint type as well. For now, LogicalCommonHint only has one hint flag: preAggEnabledByHint, and we can add more hint infos in the future if we need to.